app/org/sagebionetworks/bridge/services/SubpopulationService.java

```diff
@@ -0,0 +1,112 @@
+package org.sagebionetworks.bridge.services;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Validator;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.dao.SubpopulationDao;
+import org.sagebionetworks.bridge.models.schedules.ScheduleContext;
+import org.sagebionetworks.bridge.models.studies.Study;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
+import org.sagebionetworks.bridge.models.studies.Subpopulation;
+import org.sagebionetworks.bridge.validators.SubpopulationValidator;
+import org.sagebionetworks.bridge.validators.Validate;
+
+@Component
+public class SubpopulationService {
+
+    private SubpopulationDao subpopDao;
+    
+    @Autowired
+    final void setSubpopulationDao(SubpopulationDao subpopDao) {
+        this.subpopDao = subpopDao;
+    }
+    
+    /**
+     * Create subpopulation. 
+     * @param subpop
+     * @return
+     */
+    public Subpopulation createSubpopulation(Study study, Subpopulation subpop) {
+        checkNotNull(study);
+        checkNotNull(subpop);
+        
+        subpop.setGuid(BridgeUtils.generateGuid());
```

Comment:
Is it necessary for both the service and the DAO to set the guid? Would it be better for the service to set it and the DAO validates that it exists?